src/components.js

```diff
@@ -501,45 +441,35 @@ export function parse(root) {
  */
 export function register(name, impl) {
 
-    var F, Surrogate;
+    var definition = {};
+    var htmlProto, proto;
 
-    if (isObject(name)) {
-        impl = name;
-        name = impl.name;
-    }
+    impl = impl || {};
 
-    if (!isString(name) || !name) {
-        throw Error('"' + name + '" is not a valid component name');
+    if ('extends' in impl) {
+        htmlProto = htmlPrototypes[impl.extends];
+        definition.extends = impl.extends;
     }
-
-    if (componentClasses[name]) {
-        throw Error('A component called ' + name + ' already exists');
+    else {
+        htmlProto = HTMLElement.prototype;
+        definition.extends = 'div';
     }
 
-    impl = impl || {};
+    proto = Object.create(htmlProto);
 
-    F = function () {
-        Component.apply(this, arguments);
-    };
+    Object.keys(customElementBase).forEach(function (key) {
+        proto[key] = customElementBase[key];
+    });
```

Comment:
Have you tried something like `Object.getOwnPropertyDescriptor`?

```js
Object.keys(customElementBase).forEach(function (key) {
    let propDescriptor = Object.getOwnPropertyDescriptor(csutomElementBase, key);
    Object.defineProperty(proto, key, propDescriptor);
});
```

This will then provide the verbatim properties, along with their attributes/getters/setters.